### ISSUE TITLE
Generic virtual method pointer lookup failure in CreateParameterizedConstructor

### ISSUE THREAD

**Issue by hez2010** [dotnet/runtimelab#929]
## Generic virtual method pointer lookup failure in CreateParameterizedConstructor

Repro code: https://github.com/hez2010/SyaBot

1. `cd SyaBot.Server && dotnet publish -c Release -r win-x64`
2. Run nativeaot-built server first and then run client (`cd SyaBot.Client && dotnet run`)
3. See exception from server

Output from server:
```
Raw request string: {"Name":"hello","Data":null,"Type":"SyaBot.Shared.RegisterRequest","Id":"990e15e8-dcba-4d6d-96b8-1767365693b9"}
Request id: 990e15e8-dcba-4d6d-96b8-1767365693b9, type: SyaBot.Shared.RegisterRequest
Generic virtual method pointer lookup failure.

Declaring type handle: EEType:0x00007FF7598B48D0
Target type handle: EEType:0x00007FF7598B5398
Method name: CreateParameterizedConstructor
Instantiation:
  Argument 00000000: EEType:0x00007FF7598B6E50
  Argument 00000001: EEType:0x00007FF7598984C8
  Argument 00000002: EEType:0x00007FF759898488
  Argument 00000003: EEType:0x00007FF759898488
  Argument 00000004: EEType:0x00007FF759898488
```

Seems that it was caused by `JsonSerializer.Deserialze`, but I don't know how to resolve it:
https://github.com/hez2010/SyaBot/blob/8fbaa4ccda964b5f88fe82865d0a4fb64c6037a7/SyaBot.Server/Program.cs#L60

**Comment by jkotas** [comment]
cc @MichalStrehovsky

**Comment by MichalStrehovsky** [comment]
Yeah, I have this in my backlog, but didn't get a chance to look into it yet, sorry.

**Comment by MichalStrehovsky** [comment]
Fixed in #958. Thanks for the good repro!
